### PR TITLE
use aesara and pymc4 instead of theano and pymc3

### PR DESCRIPTION
**Pull request by AndrewAnnex**
…No clue if CI builds would work. todo need to update the github worker to allow this branch to run.

likely there are a lot of api breaking changes I did not address here, I did not attempt to run it locally yet. Also the CI environment for gempy is a bit confusing (still using travis?). Likely these scripts could be cleaned up a bit more to make testing easier

# Description
swap out all usage of theano for aesara and pymc3 for pymc4. 

Relates to <issue>
#701 

# Checklist
- [ ] My code follows the [PEP 8 style guidelines](https://www.python.org/dev/peps/pep-0008/).
- [ ] My code uses type hinting for function and method arguments and return values.
- [ ] My code contains descriptive and helpful docstrings 
which are formatted per the [Google Python Style Guidelines](http://google.github.io/styleguide/pyguide.html).
- [ ] I have created tests which entirely cover my code.
- [ ] The test code either 1. demonstrates at least one valuable use case (e.g. integration tests) 
or 2. verifies that outputs are as expected for given inputs (e.g. unit tests).
- [ ] New and existing tests pass locally with my changes.

### CODE REVIEW

**Comment by Japhiolite** [comment]
The checks were always failing at the installation step. Apparently with aesara 2.7.8 there'd been a change with setuptools (https://github.com/aesara-devs/aesara/pull/1050). Out of curiosity, I temporary froze aesara to 2.7.7 just to see whether the checks pass. Seems...like there are all tests passing now 🎉 
Will have a look at this setuptools thing to get aesara out of the ice-box.

**Comment by AndrewAnnex** [comment]
we should also do some benchmarks with a gpu to ensure things are still fast, can probably help with that

**Comment by AndrewAnnex** [comment]
@Japhiolite that bug in aesara was fixed a couple of weeks ago. I can perform some benchmarking tests if needed

File: examples/tutorials/ch5_probabilistic_modeling_DEP/ch5_2_introduction_pymc3.py
Comment (Japhiolite) [inline examples/tutorials/ch5_probabilistic_modeling_DEP/ch5_2_introduction_pymc3.py:1]: The newest arviz version ( [v0.15.1](https://python.arviz.org/en/v0.15.1/api/generated/arviz.from_pymc3.html) ) does not have a `from_pymc3()` method anymore...in fact, I don't see a `from_pymc()` method either....So I guess we'll have to see how to fix this as we go by.

**Comment by AndrewAnnex** [comment]
closing as per #796

**Pull request closed without merging**